src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -221,29 +167,32 @@ public override ExpressionSyntax VisitSwitchStatement(SwitchStatementSyntax node
                 return RewriteSwitchStatement(node);
             }
 
+            private static SwitchLabelSyntax SingleOrDefaultSwitchLabel(SyntaxList<SwitchLabelSyntax> labels)
+            {
+                return labels.Count == 1
+                    ? labels[0]
+                    : labels.First(x => x.IsKind(SyntaxKind.DefaultSwitchLabel));
+            }
+
             private ExpressionSyntax RewriteSwitchStatement(SwitchStatementSyntax node, bool allowMoveNextStatementToSwitchExpression = true)
             {
                 var switchArms = node.Sections
                     // The default label must come last in the switch expression.
-                    .OrderBy(section => section.Labels[0].IsKind(SyntaxKind.DefaultSwitchLabel))
+                    .OrderBy(section => SingleOrDefaultSwitchLabel(section.Labels).IsKind(SyntaxKind.DefaultSwitchLabel))
                     .Select(s =>
-                        (leadingTrivia: s.Labels[0].GetFirstToken().LeadingTrivia,
-                         trailingTrivia: s.Statements[0].GetLastToken().TrailingTrivia,
+                        (leadingTrivia: new[] { s.Labels[0].GetFirstToken(), s.Labels[0].GetLastToken() },
+                         trailingTrivia: new[] { s.Statements[0].GetFirstToken(), s.Statements[0].GetLastToken() } ,
                          armExpression: GetSwitchExpressionArm(s)))
                     .ToList();
 
-                // This is possibly false only on the top-level switch statement. 
-                // On nested nodes, if there's a subsequent statement, it is most definitely a
-                // "return" or "throw" which is already validated in the analysis phase.
                 if (allowMoveNextStatementToSwitchExpression)
                 {
                     var nextStatement = node.GetNextStatement();
-                    if (nextStatement != null)
+                    if (nextStatement.IsKind(SyntaxKind.ThrowStatement, SyntaxKind.ReturnStatement))
                     {
-                        Debug.Assert(nextStatement.IsKind(SyntaxKind.ThrowStatement, SyntaxKind.ReturnStatement));
                         switchArms.Add(
-                            (nextStatement.GetFirstToken().LeadingTrivia,
-                             nextStatement.GetLastToken().TrailingTrivia,
+                            (leadingTrivia: new[] { nextStatement.GetFirstToken() },
```

Comment:
Should rename the tuple elements to `tokenForLeadingTrivia` and `tokenForTrailingTrivial`